src/main/scala/vsys/blockchain/state/opcdiffs/CompareOpcDiff.scala

```diff
@@ -33,21 +33,30 @@ object CompareOpcDiff extends OpcDiffer {
     } else Left(ContractDataTypeMismatch)
 
   object CompareType extends Enumeration {
-    sealed case class CompareTypeVal(compareType: Int, op: NumComparator) extends Val(compareType) { def *(n: Int): Int = n * compareType }
-    val Ge = CompareTypeVal(1, ge)
-    val Gt = CompareTypeVal(2, gt)
-    val Le = CompareTypeVal(3, le)
-    val Lt = CompareTypeVal(4, lt)
-    val Eq = CompareTypeVal(5, _eq)
-    val Ne = CompareTypeVal(6, _ne)
+    class CompareTypeVal(compareType: Int) extends Val(compareType) { def *(n: Int): Int = n * compareType }
+    sealed case class NumCompareTypeVal(compareType: Int, op: NumComparator) extends CompareTypeVal(compareType)
+    sealed case class BytesCompareTypeVal(compareType: Int, op: (DataEntry, DataEntry) => Boolean) extends CompareTypeVal(compareType)
+    val Ge  = NumCompareTypeVal(1, ge)
+    val Gt  = NumCompareTypeVal(2, gt)
+    val Le  = NumCompareTypeVal(3, le)
+    val Lt  = NumCompareTypeVal(4, lt)
+    val Eq  = NumCompareTypeVal(5, _eq)
+    val Ne  = NumCompareTypeVal(6, _ne)
+    val Beq = BytesCompareTypeVal(7, _ == _)
+    val Bne = BytesCompareTypeVal(8, _ != _)
   }
 
-  def differ(bytes: Array[Byte], data: Seq[DataEntry], t: CompareType.CompareTypeVal) =
+  def differ(bytes: Array[Byte], data: Seq[DataEntry], t: CompareType.NumCompareTypeVal) =
     updateStack(data, bytes.last, numBiComparation(data(bytes(1)), data(bytes(2)), t.op))
 
+  def differ(bytes: Array[Byte], data: Seq[DataEntry], t: CompareType.BytesCompareTypeVal) =
+    updateStack(data, bytes.last, Right(t.op(data(bytes(1)), data(bytes(2)))))
```

Comment:
`numBiComparation` return `Either[ValidationError, DataEntry]`.
but here, `Right(t.op(data(bytes(1)), data(bytes(2))))` seems return `Right(Boolean)`.

similar in L28-31

We should use `Right(booleanToDataEntry(comparator(x, y)))` as instead, right?

Reply:
`booleanToDataEntry` is implicit function and it is `scala.language.implicitConversions` here.

Reply:
I see. I think that I checked this in the last PR. = =\\\